src/Language/DifferentialDatalog/Pos.hs

```diff
@@ -50,7 +52,24 @@ spos x = let (s,e) = pos x
          in sourceName s ++ ":" ++ (show $ sourceLine s) ++ ":" ++ (show $ sourceColumn s) ++ "-"
                                 ++ (show $ sourceLine e) ++ ":" ++ (show $ sourceColumn e)
 
-nopos::Pos 
+sposFragment :: (WithPos a) => a -> M.Map String String -> String -> String
+sposFragment element sources message =
+   let (s,e)      = pos element
+       sourceFile = sourceName s
+       lineStart  = sourceLine s
+       colStart   = sourceColumn s
+       lineEnd    = sourceLine e
+       colEnd     = sourceColumn e
+       sameLine   = lineStart == lineEnd
+       width      = if sameLine then colEnd - colStart else 1
+       line       = (lines $ (sources M.! sourceFile)) !! (lineStart - 1)
```

Comment:
This will crash if the start line is not valid, which is very much possible if there is a bug in parser or if we use artificially manufactured pos (which we do in some places). Ideally this should not happen, but the program should not crash if it does.

Also, there is a special case when position  `==nopos`, which means that we have an object whose real position is unknown or undefined (e.g., it was injected during program transformation). We should not theoretically have errors in those, but we should not crash if this happens.